### PR TITLE
added Fujitsu MB85RS4MT FRAM; added sleep mode

### PR DESCRIPTION
**Pull request by FUEL4EP**
Changes of this pull request are:

1. added Fujitsu MB85RS4MT FRAM
2. added sleep mode

The changes have been successfully verified with my PCB [HB-UNI-SEN-BATT_ATMega1284P_E07-868MS10_FRAM_FUEL4EP](https://github.com/FUEL4EP/HomeAutomation/tree/master/AsksinPP_developments/PCBs/HB-UNI-SEN-BATT_ATMega1284P_E07-868MS10_FRAM_FUEL4EP).
In sleep  mode, the current dissipation has been reduced from 30 uA (without sleep mode) to 12 uA (with sleep mode).

### CODE REVIEW

File: Adafruit_FRAM_SPI.cpp [hunk 1]
```diff
@@ -343,3 +344,36 @@ bool Adafruit_FRAM_SPI::setStatusRegister(uint8_t value) {
...
+  // Returning to an normal operation from the SLEEP mode is carried out after
+  // tREC (Max 400 μs) time from the falling edge of CS
+  spi_dev->beginTransactionWithAssertingCS();
+  delayMicroseconds(300);
+  // It is possible to return CS to H level before tREC time. However, it
+  // is prohibited to bring down CS to L level again during tREC period.
+  spi_dev->endTransactionWithDeassertingCS();
+  delayMicroseconds(100);
+
+  return spi_dev->write(&cmd, 1);
```
Comment (hathach) [inline Adafruit_FRAM_SPI.cpp]: we don't really need to do spi write() here, cs along should be able to put chip out of sleep

**Review by hathach — APPROVED** [review]
thank you for your PR, I made some changes
- add pre-commit and fix clang (quite some bit of format changes)
- rename API to enterSleep() and exitSleep()
- remove spi write in exitSleep()
- check if device support sleep mode before executing the command.

Please try it again to see if this still works for you.

**Comment by FUEL4EP** [comment]
Hello Ha Thach,

Thanks for merging my pull request. I've checked my application https://github.com/FUEL4EP/HomeAutomation/tree/master/AsksinPP_developments/sketches/HB-UNI-Sensor1-THP_MA-AHT20_BMP280 with your library commit 928281f. Everything is running smoothly as expected. In sleep mode, the current is again about 12 uA.

**Comment by FUEL4EP** [comment]
Hello Ha Thach,

Thanks for merging my pull request. I've checked my application https://github.com/FUEL4EP/HomeAutomation/tree/master/AsksinPP_developments/sketches/HB-UNI-Sensor1-THP_MA-AHT20_BMP280 with your library commit 928281f. Everything is running smoothly as expected. In sleep mode, the current is again about 12 uA.

**Comment by hathach** [comment]
great thank you for your updated feed back.

**Pull request merged**